Sample Applications/WPFGallery/ViewModels/MainWindowViewModel.cs

```diff
@@ -56,6 +89,20 @@ public MainWindowViewModel(INavigationService navigationService)
         _timer = new DispatcherTimer();
         _timer.Interval = TimeSpan.FromMilliseconds(400);
         _timer.Tick += PerformSearchNavigation;
+
+        SearchItems = new();
+        
+        foreach (ControlInfoDataItem mainPage in Controls)
```

Comment:
Even this should go in ControlsInfoDataSource.cs
Reasoning being, it is not related to MainWindow, and it may be needed by multiple viewmodels.